editor/backend/src/EditCommand.ts

```diff
@@ -80,11 +98,19 @@ export class EditCommandAdmin {
   private static _isInitialized = false;
   public static get activeCommand() { return this._activeCommand; }
 
-  public static runCommand(cmd?: EditCommand) {
-    if (this._activeCommand)
-      this._activeCommand.onFinish();
+  public static async finishCommand() {
+    if (this._activeCommand) {
+      const finished = await this._activeCommand.requestFinish();
+      if ("done" !== finished)
+        throw new BackendError(IModelStatus.ServerTimeout, editorIpcStrings.commandBusy, finished);
+    }
+    this._activeCommand = undefined;
+  }
+
+  public static async runCommand(cmd: EditCommand): Promise<any> {
```

Comment:
Are `runCommand` and `finishCommand` intended to be `@beta` now? If so please consider documenting them and other promoted APIs.

Reply:
made them @internal, but added documentation to the ipc interfaces that are used to call them.